npe2/io_utils.py

```diff
@@ -0,0 +1,201 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List, Optional, Tuple, Union, overload
+
+from typing_extensions import Literal
+
+from . import PluginManager
+from ._types import FullLayerData, LayerData, PathLike
+
+if TYPE_CHECKING:
+    from .manifest.io import ReaderContribution, WriterContribution
+
+
+def read(path: PathLike, *, plugin_name: Optional[str] = None) -> List[LayerData]:
```

Comment:
I'd prefer `read_layer_data`.  Something longer is easier to search for and it seems this function and `write_layer_data` should have similar names.

Reply:
Do we ever expect to be reading writing anything besides layer data here?  If not, feels like npe2.read and npe2.write are much nicer.  We can always add an alias though and both be happy :)

Reply:
Based on the types, these functions look specific to layer data.  And we don't do anything more general right now, so I lean towards the more specific longer name.

I do like short verbs. I think I'd lean more towards the short name if this were going to be used on the command line more, but usually these will get invoked inside napari.

Really, either way is fine as long as the names are self consistent: both short or both long.

Reply:
> but usually these will get invoked inside napari.

no, that's kind of the whole point!  As we've discussed a few times, I'm specifically seeing these functions as the solution to https://github.com/napari/napari-plugin-engine/issues/24

Reply:
> Based on the types, these functions look specific to layer data. And we don't do anything more general right now

... and I can't imagine we ever would.  It's that, combined with my desire to finally solve napari/napari-plugin-engine#24 that's making me suggest this.  (if it were just for napari to use, I wouldn't really care at all what we call it)

Reply:
> I can't imagine we ever would.

oh that's interesting.  The "more" that I'm thinking of is something like the napari configuration: settings + selected ui state + layer data.  But, if I'm understanding, that would be very napari specific and not done in this plugin engine.

Reply:
> But, if I'm understanding, that would be very napari specific and not done in this plugin engine.

oh yeah def.  napari uses npe2, npe2 doesn't use napari.  npe2 will likely never know about UI state

Reply:
alright I'm sold on the short names.